modules/images/image-loading-optimization/storage/data.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * Metrics storage data.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets the freshness age (TTL) for a given page metric.
+ *
+ * When a page metric expires it is eligible to be replaced by a newer one if its viewport lies within the same breakpoint.
+ *
+ * @return int Expiration TTL in seconds.
+ */
+function ilo_get_page_metric_freshness_ttl() {
+	/**
+	 * Filters the freshness age (TTL) for a given page metric.
+	 *
+	 * @param int $ttl Expiration TTL in seconds.
+	 */
+	return (int) apply_filters( 'ilo_page_metric_freshness_ttl', DAY_IN_SECONDS );
+}
+
+/**
+ * Gets the normalized query vars for the current request.
+ *
+ * This is used as a cache key for stored page metrics.
+ *
+ * @return array Normalized query vars.
+ */
+function ilo_get_normalized_query_vars() {
+	global $wp;
+
+	// Note that the order of this array is naturally normalized since it is
+	// assembled by iterating over public_query_vars.
+	$normalized_query_vars = $wp->query_vars;
+
+	// Normalize unbounded query vars.
+	if ( is_404() ) {
+		$normalized_query_vars = array(
+			'error' => 404,
+		);
+	} elseif ( array_key_exists( 's', $normalized_query_vars ) ) {
+		$normalized_query_vars['s'] = '...';
```

Comment:
Just so I better understand: So it doesn't matter when a search results page sometimes shows only 1 post with no image, and sometimes it shows a post with an LCP image?

Reply:
Oh yes, you're absolutely right. That is problematic. The goal behind normalization is to prevent filling up the `ilo_page_metric` posts with garbage. But it is certainly possible that different search results will result in a random assortment of posts that do and do not have featured images. 

Indeed, this will also be a problem for archive pages in general. One way to deal with this would be to include the post IDs in The Loop among the query vars so that a new post being published would effectively invalidate the cache for page metrics. But I don't think this makes sense to do for search results. Perhaps image loading optimization should just be disabled for search results. What do you think?

Reply:
I've disabled optimizing search results entirely in 99ef2996caf805cf2ca0d253374ee66e901cbf84 and added a `ilo_can_optimize_response` filter which can be used to override which responses get optimized.

Reply:
Sounds like a good compromise 👍